python/ml4ir/applications/classification/model/classification_model.py

```diff
@@ -103,6 +93,56 @@ def evaluate(
             self.logger.info(f"Evaluation Results written at: {logs_dir}")
         return global_metrics, grouped_metrics, metrics_dict
 
+    @staticmethod
+    def get_chunks_from_df(dataframe, size):
+        """
+        Given a pd.DataFrame, creates batches of size `size`.
+        """
+        return [dataframe.iloc[pos:pos + size] for pos in range(0, len(dataframe), size)]
```

Comment:
Just out of curiosity (may be a dumbish question), would we benefit to use a generator for our use-case here?

Reply:
I think we should use generators if possible. I have done the change.